src/core/qgstaskmanager.cpp

```diff
@@ -142,6 +142,27 @@ QList<QgsMapLayer *> QgsTask::dependentLayers() const
   return _qgis_listQPointerToRaw( mDependentLayers );
 }
 
+bool QgsTask::waitForFinished( int timeout )
```

Comment:
Is the intention with pause to get this task to run faster? Because task pausing only works for queued tasks, not running tasks....